systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectBuilderST.java

```diff
@@ -0,0 +1,328 @@
+                .withName("connector-with-empty-checksum")
+                .withArtifacts(new JarArtifactBuilder()
+                    .withNewUrl("https://github.com/scholzj/echo-sink/releases/download/1.1.0/echo-sink-1.1.0.jar")
+                    .withNewSha512sum("b7da48d5ecd1e4199886d169ced1bf702ffbdfd704d69e0da97e78ff63c1bcece2f59c2c6c751f9c20be73472b8cb6a31b6fd4f75558c1cb9d96daa9e9e603d2")
+                    .build())
+                .build();
+
+            kafkaConnect.getSpec().getBuild().getPlugins().remove(0);
+            kafkaConnect.getSpec().getBuild().getPlugins().add(pluginWithRightChecksum);
+        });
+
+        KafkaConnectUtils.waitForConnectReady(clusterName);
+    }
+
+    @Test
+    void testBuildWithJarAndTgz() {
+        // this test also testing push into Docker output
+        String topicName = KafkaTopicUtils.generateRandomNameOfTopic();
+
+        KafkaTopicResource.create(KafkaTopicResource.topic(clusterName, topicName).build());
+
+        KafkaConnectResource.create(KafkaConnectResource.kafkaConnect(clusterName, 1)
+            .editMetadata()
+                .addToAnnotations(Annotations.STRIMZI_IO_USE_CONNECTOR_RESOURCES, "true")
+            .endMetadata()
+            .editOrNewSpec()
+                .addToConfig("key.converter.schemas.enable", false)
+                .addToConfig("value.converter.schemas.enable", false)
+                .addToConfig("key.converter", "org.apache.kafka.connect.storage.StringConverter")
+                .addToConfig("value.converter", "org.apache.kafka.connect.storage.StringConverter")
+                .withNewBuild()
+                    .withPlugins(PLUGIN_WITH_TAR_AND_JAR)
+                    .withNewDockerOutput()
+                        .withNewImage(imageName)
+                    .endDockerOutput()
+                .endBuild()
+                .withNewInlineLogging()
+                    .addToLoggers("connect.root.logger.level", "INFO")
+                .endInlineLogging()
+            .endSpec()
+            .build(), false);
+
+        Map<String, Object> connectorConfig = new HashMap<>();
+        connectorConfig.put("topics", topicName);
+        connectorConfig.put("level", "INFO");
+
+        KafkaClientsResource.create(KafkaClientsResource.deployKafkaClients(false, kafkaClientsName).build());
+        String kafkaClientsPodName = kubeClient().listPodsByPrefixInName(kafkaClientsName).get(0).getMetadata().getName();
+
+        KafkaConnectorResource.create(KafkaConnectorResource.kafkaConnector(clusterName)
+            .editOrNewSpec()
+                .withClassName(ECHO_SINK_CLASS_NAME)
+                .withConfig(connectorConfig)
+            .endSpec()
+            .build());
+
+        KafkaConnector kafkaConnector = KafkaConnectorResource.kafkaConnectorClient().inNamespace(NAMESPACE).withName(clusterName).get();
+
+        assertThat(kafkaConnector.getSpec().getClassName(), is(ECHO_SINK_CLASS_NAME));
```

Comment:
Should this use some _waitForReady_ or something like that? It looks a bit weird that you just create the resource, then get it and assert it.

Reply:
The `waitForReady` method is inside the `create` method, so we are waiting for the readiness and then just get it and do the assertion

Reply:
Ok ... for another PR ... but maybe calling it `createAndWaitFOrReadiness` or something would improve the code readability.

Reply:
Okay, that sounds good, I know that it's not quite clear what the `create` method does. Thanks :slightly_smiling_face: